### PR TITLE
create script to generate vendored modules tarball for etcd and coredns

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

create script to generate vendored modules tarball for etcd and coredns

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/coredns/generate_source_tarball.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# Licensed under the MIT License.
+
+# Quit on failure
+set -e
+
+#
+# The nodejs source tarball contains a copy of the OpenSSL source tree.
+# OpenSSL contains patented algorithms that should not be distributed
+# as part of the SRPM. Since we use the shared OpenSSL libraries, we 
+# can just remove the entire OpenSSL source tree from the tarball.
```
Comment (oliviacrain) [inline SPECS/coredns/generate_source_tarball.sh:12]: Comment not applicable to this file

File: SPECS/etcd/generate_source_tarball.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+# Licensed under the MIT License.
+
+# Quit on failure
+set -e
+
+#
+# The nodejs source tarball contains a copy of the OpenSSL source tree.
+# OpenSSL contains patented algorithms that should not be distributed
+# as part of the SRPM. Since we use the shared OpenSSL libraries, we 
+# can just remove the entire OpenSSL source tree from the tarball.
```
Comment (oliviacrain) [inline SPECS/etcd/generate_source_tarball.sh:12]: Comment not applicable to this script

File: SPECS/coredns/generate_source_tarball.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+echo "Tar vendored modules"
+tar  --sort=name \
+     --mtime="2021-04-26 00:00Z" \
+     --owner=0 --group=0 --numeric-owner \
+     --pax-option=exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime \
+     -cf "$VENDOR_TARBALL" vendor
+
+popd > /dev/null
+echo "Coredns vendore modules are available at $VENDOR_TARBALL"
```
Comment (oliviacrain) [inline SPECS/coredns/generate_source_tarball.sh:104]:
```suggestion
echo "Coredns vendored modules are available at $VENDOR_TARBALL"
```

**Review by oliviacrain — APPROVED** [review]

File: SPECS/coredns/generate_source_tarball.sh (hunk 1 above)
Comment (nicogbg) [reply SPECS/coredns/generate_source_tarball.sh:12]: oops, I'll remove it

**Pull request merged**